v2/telemetry.js

```diff
@@ -268,114 +330,165 @@ Telemetry.init = function Telemetry_init(callback) {
-      callback({});
-    } else {
-      filterOptions["metric"] = filterOptions["metric"].filter(function(measure) {
-        return !/^STARTUP_/.test(measure); // Ignore STARTUP_* histograms since nobody ever uses them
-      });
-      callback(filterOptions);
+  Telemetry.getJSON(
+    Telemetry.BASE_URL + "filters/?channel=" + encodeURIComponent(channel) +
+    "&version=" + encodeURIComponent(version),
+    function(filterOptions, status) {
+      if (filterOptions === null) {
+        // Only allow no filter options if it is 404 - if there are no filters
+        assert(status === 404, "Could not obtain filter options: status " + status + " (" + url + ")");
+        callback({});
+      } else {
+        filterOptions["metric"] = filterOptions["metric"].filter(function(measure) {
+          return !/^STARTUP_/.test(measure); // Ignore STARTUP_* histograms since nobody ever uses them
+        });
+        callback(filterOptions);
+      }
     }
-  });
+  );
 }
 
 Telemetry.getVersions = function Telemetry_getVersions(fromVersion, toVersion) { // shim function
   assert(Telemetry.CHANNEL_VERSION_DATES !== null && Telemetry.CHANNEL_VERSION_BUILDIDS !== null, "Telemetry.js must be initialized before use");
-  assert((fromVersion === undefined && toVersion === undefined) || (typeof fromVersion === "string" && typeof toVersion === "string"), "`fromVersion` and `toVersion` must be strings");
+  assert(
+    (fromVersion === undefined && toVersion === undefined) ||
+    (typeof fromVersion === "string" && typeof toVersion === "string"),
+    "`fromVersion` and `toVersion` must be strings or omitted"
+  );
   var versions = [];
   for (var channel in Telemetry.CHANNEL_VERSION_DATES) {
     for (var version in Telemetry.CHANNEL_VERSION_DATES[channel]) {
-      // Only output a channel/version if it is in both the buildID versions and the submission date versions (sometimes there is invalid data where a version is in submission date versions but not buildID versions, see https://bugzilla.mozilla.org/show_bug.cgi?id=1189727)
-      if (Telemetry.CHANNEL_VERSION_BUILDIDS[channel] && Telemetry.CHANNEL_VERSION_BUILDIDS[channel][version]) {
-        versions.push(channel + "/" + version);
+      // Only output a channel/version if it is in both the buildID versions and
+      // the submission date versions (sometimes there is invalid data where a version
+      // is in submission date versions but not buildID versions,
+      // see https://bugzilla.mozilla.org/show_bug.cgi?id=1189727 for details)
+      if (Telemetry.CHANNEL_VERSION_BUILDIDS[channel]) {
+        if (Telemetry.CHANNEL_VERSION_BUILDIDS[channel][version]) {
+          versions.push(channel + "/" + version);
+        }
       }
     }
   }
+
   versions.sort();
-  return fromVersion !== undefined ? versions.filter(function(version) { return fromVersion <= version && version <= toVersion }) : versions;
+  return fromVersion !== undefined ? versions.filter(function(version) {
```

Comment:
this return is way too fancy. break it up with some if's and temporary variables